src/components/networked-video-source.js

```diff
@@ -0,0 +1,87 @@
+/* global AFRAME, NAF, THREE */
+var naf = require('../NafIndex');
+
+AFRAME.registerComponent('networked-video-source', {
+
+  schema: {
+  },
+
+  dependencies: ['material'],
+
+  init: function () {
+    this.videoTexture = null;
+    this.video = null;
+    this.stream = null;
+
+    this._setMediaStream = this._setMediaStream.bind(this);
+
+    NAF.utils.getNetworkedEntity(this.el).then((networkedEl) => {
+      const ownerId = networkedEl.components.networked.data.owner;
+
+      if (ownerId) {
+        NAF.connection.adapter.getMediaStream(ownerId, "video")
+          .then(this._setMediaStream)
+          .catch((e) => naf.log.error(`Error getting media stream for ${ownerId}`, e));
+      } else {
+        // Correctly configured local entity, perhaps do something here for enabling debug audio loopback
+      }
+    });
+  },
+
+  _setMediaStream(newStream) {
+
+    if(!this.video) {
+      this.setupVideo();
+    }
+
+    if(newStream != this.stream) {
+      if (this.stream) {
+        this._clearMediaStream();
+      }
+
+      if (newStream) {
+        this.video.srcObject = newStream;
+
+        var playResult = this.video.play();
+        if (playResult instanceof Promise) {
+          playResult.catch((e) => naf.log.error(`Error play video stream`, e));
+        }
+
+        const mesh = this.el.getObject3D('mesh');
+        mesh.material.map = new THREE.VideoTexture(this.video);
+        mesh.material.needsUpdate = true;
+      }
+
+      this.stream = newStream;
+    }
+  },
+
+  _clearMediaStream() {
+    if (this.video) {
+      this.video.srcObject = null;
+      this.video = null;
+      this.stream = null;
+    }
+  },
+
+  remove: function() {
+    if (!this.videoTexture) return;
```

Comment:
`this.videoTexture` is not currently set.
We may explicitly dispose the texture too. See https://github.com/mozilla/hubs/blob/fef69c13341ec148211c53b3b5d6f2790519ec65/src/components/video-texture-target.js#L105

Reply:
will do.